src/gtk/window.cpp

```diff
@@ -1429,6 +1428,20 @@ gtk_window_key_press_callback( GtkWidget *WXUNUSED(widget),
                 // use Unicode values
                 eventChar.m_keyCode = key_code;
                 eventChar.m_uniChar = uniChar;
+
+            } else if (event.ControlDown()) {
+
+                if (( uniChar < 'a' || uniChar > 'z' ) &&
+                    ( uniChar < 'A' || uniChar > 'Z' )) {
+
+                    // for Ctrl+NonLatinLetters use Unicode values also
+                    eventChar.m_uniChar = uniChar;
+
+                } else {
+
+                    // use keycode (1..26) value for Unicode field to mimic MSW behavior
+                    eventChar.m_uniChar = eventChar.m_keyCode;
+                }
```

Comment:
Would it be possible to keep this inside `AdjustCharEventKeyCodes()`? It's difficult to understand this logic when it's split between two places...